### PR TITLE
Switch to pkgdown Bootstrap 5 template

### LINKED ISSUE #5503: Switch to pkgdown Bootstrap 5 template?
The switch to bs5 will allow for search within the site, which will be very useful for users.
https://www.tidyverse.org/blog/2021/12/pkgdown-2-0-0/#bootstrap-5

Since you are currently using pkgdown 2 with CI, switching to bs5 should be very easy.
(Of course I could create a PR if you are OK)

### PR DESCRIPTION
**Pull request by eitsupi**
Close #5503

The switch to bs5 will allow for search within the site, which will be very useful for users.

It will look like the following and a search window will be added in the upper right corner.

image

This PR is a small change to the pkgdown site configuration only; there are no changes to the document itself or to the R code.

### How to test?

1. Pull this branch.
2. Run the `pkgdown::build_site()` function on the project root.
3. The website code will be generated in the ./docs dir.

### CODE REVIEW

**Comment by eitsupi** [comment]
@jangorecki Could you take a look at this?

**Comment by jangorecki** [comment]
LGTM

**Comment by jangorecki** [comment]
Any idea why this is not the default in pkgdown?

File: _pkgdown.yml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 url: https://rdatatable.gitlab.io/data.table
 
+template:
+  bootstrap: 5
+  params:
+    ganalytics: UA-129166154-2
```
Comment (jangorecki) [inline _pkgdown.yml:6]: we don't have those in master already, should have gone during git merge master

**Comment by eitsupi** [comment]
> Any idea why this is not the default in pkgdown?

This is a breaking change and we just need to opt in, check the pkgdown release notes.

File: _pkgdown.yml (hunk 1 above)
Comment (eitsupi) [inline _pkgdown.yml:6]:
```suggestion
```

**Comment by eitsupi** [comment]
Anyway, this is not a major change and I see no reason why it cannot be merged immediately.
At the moment the documentation for data.table is very difficult for beginners to read as it is not searchable.

**Comment by jangorecki** [comment]
Website looks much better on the mobile browser now

**Comment by eitsupi** [comment]
Great! Thanks for merging this.

**Pull request merged**